api/controllers/UserController.js

```diff
@@ -100,8 +100,36 @@ module.exports = {
   },
 
   logout: function (req, res) {
-    delete req.session.usr;
-    req.session.loggedIn = false;
-    return res.ok();
+    // If the user isn't logged in, just get them out of here
+    if (!req.session.loggedIn) {
+      return res.ok();
+    }
+    // https://github.com/expressjs/session#sessiondestroycallback
+    req.session.destroy(function afterDestroy() {
+      return res.ok();
+    });
+  },
+
+  status: async function (req, res) {
+    const { usr: id, loggedIn: authenticated } = req.session;
+
+    // User is not logged in, get out of here
+    if (!authenticated || !id) {
+      return res.ok({
+        authenticated: false,
+      });
+    }
+
+    try {
+      // If the user is logged in, see if we can find them first to verify they exist
+      const { username } = await userAPI.findUser(id);
+      return res.ok({
+        id,
+        username,
+        authenticated,
+      });
+    } catch (err) {
+      return res.badRequest(err);
+    }
```

Comment:
Should we log you out in this case? Like if your session says you're user id 1337, but we can't find that user, maybe we should clear the session data so you can reauthenticate as a real user?